modules/data.remote/R/remote_process.R

```diff
@@ -0,0 +1,703 @@
+##' call rp_control (from RpTools py package) from PEcAn workflow and store the output in BETY
+##'
+##' @name remote_process
+##' @title remote_process
+##' @export
+##' @param settings PEcAn settings list containing remotedata tags: raw_mimetype, raw_formatname, geofile, source, collection, scale, projection, qc, algorithm, credfile, pro_mimetype, pro_formatname, out_get_data, out_process_data
+##' @examples
+##' \dontrun{
+##' remote_process(settings)
+##' }
+##' @author Ayush Prasad
+##'
+
+remote_process <- function(settings) {
+  # information about the date variables used in remote_process -
+  # req_start, req_end : start, end dates requested by the user, the user does not have to be aware about the status of the requested file in the DB
+  # start, end : effective start, end dates created after checking the DB status. These dates are sent to remote_process for downloading and processing data
+  # write_raw_start, write_raw_end : start, end dates which are used while inserting and updating the DB
+  # the "pro" version of these variables have the same meaning and are used to refer to the processed file
+  
+  RpTools <- reticulate::import("RpTools")
+  
+  input_file <- NULL
+  stage_get_data <- NULL
+  stage_process_data <- NULL
+  raw_merge <- NULL
+  pro_merge <- NULL
+  existing_raw_file_path <- NULL
+  existing_pro_file_path <- NULL
+  
+  # extract the variables from the settings list
+  siteid <- as.numeric(settings$run$site$id)
+  siteid_short <- paste0(siteid %/% 1e+09, "-", siteid %% 1e+09)
+  raw_mimetype <- settings$remotedata$raw_mimetype
+  raw_formatname <- settings$remotedata$raw_formatname
+  outdir <- settings$outdir
+  start <- as.character(as.Date(settings$run$start.date))
+  end <- as.character(as.Date(settings$run$end.date))
+  source <- settings$remotedata$source
+  collection <- settings$remotedata$collection
+  scale <- settings$remotedata$scale
+  if (!is.null(scale)) {
+    scale <- as.double(settings$remotedata$scale)
+    scale <- format(scale, nsmall = 1)
+  }
+  projection <- settings$remotedata$projection
+  qc <- settings$remotedata$qc
+  if (!is.null(qc)) {
+    qc <- as.double(settings$remotedata$qc)
+    qc <- format(qc, nsmall = 1)
+  }
+  algorithm <- settings$remotedata$algorithm
+  credfile <- settings$remotedata$credfile
+  pro_mimetype <- settings$remotedata$pro_mimetype
+  pro_formatname <- settings$remotedata$pro_formatname
+  out_get_data <- settings$remotedata$out_get_data
+  out_process_data <- settings$remotedata$out_process_data
+  
+  
+  dbcon <- PEcAn.DB::db.open(settings$database$bety)
+  on.exit(PEcAn.DB::db.close(dbcon), add = TRUE)
+  
+  flag <- 0
```

Comment:
can this be renamed to be a bit more informative, e.g. `remotefile_check_flag` or something like that? and could you document the meanings of each number in a single place, maybe as inline comments here (wherever it is used for the first time)